src/java_bytecode/generate_java_generic_method.h

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************\
+
+Module: Generate Java Generic Method - Instantiate a generic method
+        with concrete type information.
+
+Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#ifndef CPROVER_JAVA_BYTECODE_GENERATE_JAVA_GENERIC_METHOD_H
+#define CPROVER_JAVA_BYTECODE_GENERATE_JAVA_GENERIC_METHOD_H
+
+#include <util/message.h>
+#include <util/symbol_table.h>
+#include <util/std_types.h>
+#include <java_bytecode/java_types.h>
+
+class generate_java_generic_methodt
+{
+public:
+  typedef std::initializer_list
+    <std::pair<const symbol_typet, const symbol_typet>>
+      type_variable_instantiationst;
+
+  typedef std::unordered_map
+    <const symbol_typet, const symbol_typet, irep_hash>
```

Comment:
I think there is a small danger in using the `irep_hash` here is if someone adds something to a `symbol_typet` this might do something unexpected. I think using the identifier directly (e.g. a map from ``irep_idt` -> `symbol_typet`)

Reply:
I was thinking that in most cases where you want to look up something in the map, you have a `symbol_typet` on hand so you may as well look up using that as the key - but I take you point that it creates an extra dependency on the behaviour of `symbol_typet`, so I'll make this switch, thanks.